modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/authcode/UpdatingAuthorizationCodeState.java

```diff
@@ -55,14 +66,29 @@ private void update(ResourceOwnerOAuthContext context) {
         onUpdate.accept(context);
       }
     });
+    this.tokenSupplier = () -> {
+      try {
+        return valueOf(dancer.accessToken(getResourceOwnerId()));
+      } catch (RequestAuthenticationException e) {
+        throw new RuntimeException(e);
+      }
+    };
+    this.isClusterEnabled = isClusterEnabled;
   }
 
   @Override
   public String getAccessToken() {
     if (invalidated) {
-      throw new TokenInvalidatedException(
-                                          "OAuth token for resource owner id " + delegate.getResourceOwnerId()
-                                              + " has been invalidated");
+      throw new TokenInvalidatedException(format(INVALIDATED_MESSAGE_TEMPLATE, getResourceOwnerId()));
+    }
+
+    if (isClusterEnabled) {
+      try {
+        tokenSupplier.get();
+      } catch (Exception e) {
+        invalidated = true;
```

Comment:
just because this failed it doesn't mean the token is invalidated, specially if the meaning of `invalidated` is `invalidated by user`